File: problems_list/剑指OfferII074.合并区间/剑指OfferII074-合并区间.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a, b)=>a[0]-b[0]);
    let st=-1, ed=-1;
    let ans=[];
    for(const [s, e] of intervals){
        if(ed<s){
            if(st!==-1) ans.push([st, ed]);
            st=s, ed=e;
        }else{
            ed=Math.max(ed, e);
        }
    }
    if(st!==-1) ans.push([st, ed]);
    return ans;
};

File: problems_list/剑指OfferII074.合并区间/剑指OfferII074-合并区间.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (intervals.size() == 0) {
            return {};
        }
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> merged;
        for (int i = 0; i < intervals.size(); ++i) {
            int L = intervals[i][0], R = intervals[i][1];
            if (!merged.size() || merged.back()[1] < L) {
                merged.push_back({L, R});
            }
            else {
                merged.back()[1] = max(merged.back()[1], R);
            }
        }
        return merged;
    }
};